### PR TITLE
Feature wallet pocket

### PR DESCRIPTION
**Pull request by sowen123**

### CODE REVIEW

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 This page describes changes to the Service Interface Base API implemented across different releases of the interface.
+## 3.32.0
+Released 14 June 2021
+* Added a new `WalletPocket` model to store a name and id of a wallet pocket if it has one.
```
Comment (TimElectrum) [inline src/docs/devguide/hugo/site/content/release-notes.md:4]:
```suggestion
* Added a new `WalletPocket` model to store a name and ID of a wallet pocket if it has one.
```

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 2]
```diff
@@ -1,4 +1,9 @@
...
+* Added a `WalletPocket` field to `WalletPayment` to describe the pocket from which the payment should be made from 
+  (if the payment can be made from more than 1 pocket).
```
Comment (TimElectrum) [inline src/docs/devguide/hugo/site/content/release-notes.md:6]: - Too many "from"s
- If we're using markdown I'm happy to let the viewer decide where to wrap it

```suggestion
* Added a `WalletPocket` field to `WalletPayment` to describe the pocket from which the payment should be made (if the payment can be made from more than 1 pocket).
```

**Review by TimElectrum — COMMENTED** [review]
I made a quick submit to `master` to fix the indentation cos it was hiding your actual changes. Please pull from `master` again and ping me and I'll take a further look at this for you

File: src/test/java/io/electrum/vas/model/NewModelTests.java [hunk 3]
```diff
@@ -1,6 +1,13 @@
 package io.electrum.vas.model;
```
Comment (TimElectrum) [inline src/test/java/io/electrum/vas/model/NewModelTests.java:1]: please add to the other tests (de/serialisation etc.). Right now I'd tell you that's "just for completeness" but one day those tests are actually going to catch a bug. But the test you don't write never catches a bug

**Review by TimElectrum — CHANGES_REQUESTED** [review]
I feel like we spoke about so much more - are these the only changes we need?

File: src/test/java/io/electrum/vas/model/NewModelTests.java (hunk 3 above)
Comment (sowen123) [reply src/test/java/io/electrum/vas/model/NewModelTests.java:1]: Do you mean add a `WalletPocket` to those tests? I updated the `WalletPayment` to include a wallet pocket but should I test the `WalletPocket` separately?

File: src/docs/devguide/hugo/site/content/release-notes.md (hunk 1 above)
Comment (sowen123) [reply src/docs/devguide/hugo/site/content/release-notes.md]: changed

File: src/docs/devguide/hugo/site/content/release-notes.md (hunk 2 above)
Comment (sowen123) [reply src/docs/devguide/hugo/site/content/release-notes.md]: changed

File: src/main/java/io/electrum/vas/model/WalletPayment.java [hunk 4]
```diff
@@ -33,6 +35,28 @@ public String getWalletId() {
       return walletId;
    }
 
+   /**
+    * The pocket associated with this wallet from which payments can be made
```
[the remark this answers is not in the document]
Comment (sowen123) [reply src/main/java/io/electrum/vas/model/WalletPayment.java]: changed this wording to make it more descriptive

File: src/main/java/io/electrum/vas/model/WalletPocket.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package io.electrum.vas.model;
+
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.swagger.annotations.ApiModelProperty;
+
+public class WalletPocket {
```
[the remark this answers is not in the document]
Comment (sowen123) [reply src/main/java/io/electrum/vas/model/WalletPocket.java:15]: I'm not sure if my java doc makes sense please check me?

File: src/main/java/io/electrum/vas/model/WalletPayment.java [hunk 6]
```diff
@@ -33,6 +35,28 @@ public String getWalletId() {
...
+   public WalletPocket getWalletPocket() {
+      return walletPocket;
+   }
+
+   public void setWalletPocket(WalletPocket walletPocket) {
+      this.walletPocket = walletPocket;
+   }
+
    @Override
    public boolean equals(Object o) {
```
[the remark this answers is not in the document]
Comment (sowen123) [reply src/main/java/io/electrum/vas/model/WalletPayment.java:65]: fixed

File: src/main/java/io/electrum/vas/model/WalletPocket.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+   @ApiModelProperty(value = "A programmatic ID that can be used to identify this pocket when the name is not enough")
+   @JsonProperty("pocketId")
+   public String getPocketId() {
+      return pocketId;
+   }
+
+   public void setPocketId(String pocketId) {
+      this.pocketId = pocketId;
+   }
```
[the remark this answers is not in the document]
Comment (sowen123) [reply src/main/java/io/electrum/vas/model/WalletPocket.java:63]: added

File: src/test/java/io/electrum/vas/model/NewModelTests.java [hunk 8]
```diff
@@ -281,8 +282,12 @@ public void testRecursiveValidationOnSubFields(Object objectWithInvalidSubField,
                               .merchantType("1234")
                               .merchantName(new MerchantName().city("cpt").name("name").country("ZA").region("ZA")))
               },
-              {new WalletPayment().walletId(null).amount(new LedgerAmount().amount(456L).currency(null)),
-                      new WalletPayment().walletId("0712345678").amount(new LedgerAmount().amount(456L).currency("710"))},
+              {new WalletPayment().walletId(null).walletPocket(new WalletPocket())
+                      .amount(new LedgerAmount().amount(456L).currency(null)),
+                      new WalletPayment().walletId("0712345678")
+                              .walletPocket((new WalletPocket()).pocketName("Cash Pocket"))
```
[the remark this answers is not in the document]
Comment (sowen123) [reply src/test/java/io/electrum/vas/model/NewModelTests.java:288]: removed

**Comment by sowen123** [comment]
> I feel like we spoke about so much more - are these the only changes we need?

Yes from what we discussed this was the only change that needed to be made to the base. Unless you still want to add `AccountType` to an `AccountPaymentt` but I thought we decided to do this instead

File: src/test/java/io/electrum/vas/model/NewModelTests.java (hunk 3 above)
Comment (TimElectrum) [reply src/test/java/io/electrum/vas/model/NewModelTests.java:1]: oh right. No, that's fine

File: src/test/java/io/electrum/vas/model/NewModelTests.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (TimElectrum) [reply src/test/java/io/electrum/vas/model/NewModelTests.java:288]: really?

**Review by TimElectrum — CHANGES_REQUESTED** [review]
Github thinks those brackets are actually still lurking

**Review by TimElectrum — APPROVED** [review]

**Pull request merged**